test/mmu_test.c

```diff
@@ -0,0 +1,61 @@
+#include <criterion/criterion.h>
+#include <criterion/new/assert.h>
+#include <criterion/logging.h>
+#include <criterion/redirect.h>
+#include <signal.h>
+#include <libgen.h>
+
+#include "mmu.h"
+#include "rom.h"
+#include "log.h"
+
+#define MAX_PATH_LENGTH    (512)
+#define MAX_LOG_MSG_LENGTH (512)
+
+Test(mmu, mmu_init_with_rom_load, .exit_code = EXIT_SUCCESS) {
+    char *file_path_copy                = strdup(__FILE__);
+    char rom_file_path[MAX_PATH_LENGTH] = {0};
+    snprintf(rom_file_path, MAX_PATH_LENGTH, "%s/../roms/yobemag.gb", dirname(file_path_copy));
+    rom_init(rom_file_path);
+    mmu_init();
```

Comment:
maybe assert that the boot rom was successfully copied over into our memory

Reply:
I was thinking about doing that. What stopped me from doing it was to make the `boot_rom` global or something.

What could be done, although not really elegant since it creates 2 copies, is to have a local copy of the boot rom for the tests. This could be compared with `mem` by looping over it.

Reply:
Another possibility would be to check samples of the boot ROM, e.g. every 2-potency or something along those lines